WordPress/Classes/ViewRelated/Likes/LikesListController.swift

```diff
@@ -0,0 +1,279 @@
+        }, failure: { [weak self] _ in
+            // TODO: Show error state?
+            self?.isLoadingContent = false
+        })
+    }
+
+    /// Convenient method that fetches likes data depending on the notification's content type.
+    /// - Parameters:
+    ///   - success: Closure to be called when the fetch is successful.
+    ///   - failure: Closure to be called when the fetch failed.
+    private func fetchLikes(success: @escaping ([RemoteUser]?) -> Void, failure: @escaping (Error?) -> Void) {
+        switch content {
+        case .post(let postID):
+            dependency.postService.getLikesForPostID(postID,
+                                                     siteID: siteID,
+                                                     success: success,
+                                                     failure: failure)
+        case .comment(let commentID):
+            dependency.commentService.getLikesForCommentID(commentID,
+                                                           siteID: siteID,
+                                                           success: success,
+                                                           failure: failure)
+        }
+    }
+}
+
+// MARK: - Table View Related
+
+extension LikesListController: UITableViewDataSource, UITableViewDelegate {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return Constants.numberOfSections
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // header section
+        if section == Constants.headerSectionIndex {
+            return Constants.numberOfHeaderRows
+        }
+
+        // users section
+        return isLoadingContent ? Constants.numberOfLoadingRows : likingUsers.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.section == Constants.headerSectionIndex {
+            return headerCell()
+        }
+
+        if isLoadingContent {
+            return loadingCell
+        }
+
+        return userCell(for: indexPath)
+    }
+
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return Constants.estimatedRowHeight
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        if indexPath.section == Constants.headerSectionIndex {
+            delegate?.didSelectHeader()
+            return
+        }
+
+        guard !isLoadingContent, indexPath.row < likingUsers.count else {
+            return
+        }
+
+        delegate?.didSelectUser(likingUsers[indexPath.row])
+    }
+
+}
+
+// MARK: - Notification Cell Handling
+
+extension LikesListController {
```

Comment:
I think this can be a `private` extension since everything in it should be `private`.